taskvine/src/manager/vine_manager.c

```diff
@@ -2723,6 +2728,11 @@ static void commit_task_to_worker(struct vine_manager *q, struct vine_worker_inf
 	if (result != VINE_SUCCESS) {
 		debug(D_VINE, "Failed to send task %d to worker %s (%s).", t->task_id, w->hostname, w->addrport);
 		handle_failure(q, w, t, result);
+	} else {
+		if (t->needs_library) {
+			struct vine_task *library_task = hash_table_lookup(w->libraries, t->needs_library);
```

Comment:
Use `vine_manager_find_library_on_worker` to find **one** of the running libraries on the worker with an available slot, and then store it in `t->library_task` so that you can correctly un-do the binding when the task ends.